GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -2933,6 +2946,11 @@ private void toggleBranchTreePanel_Click(object sender, EventArgs e)
                 new Dictionary<string, string> { { "ShowLeftPanel", MainSplitContainer.Panel1Collapsed.ToString() } });
 
             RefreshLayoutToggleButtonStates();
+
+            if (!MainSplitContainer.Panel1Collapsed)
+            {
+                RefreshLeftPanel(forceRefresh: true, new FilteredGitRefsProvider(UICommands.GitModule).GetRefs);
```

Comment:
Refs are loaded again, but that is OK, too much hassle to save the data

Reply:
You're right, but I don't think showing/hiding the left panel is something users do all the time (compared to refreshing the grid). So it shouldn't cause a lot of perf degradation.